lifecycle.go

```diff
@@ -555,7 +555,9 @@ func (p *DockerProvider) preCreateContainerHook(ctx context.Context, req Contain
 	}
 	req.HostConfigModifier(hostConfig)
 
-	if req.EnpointSettingsModifier != nil {
+	if req.EndpointSettingsModifier != nil {
```

Comment:
suggestion: let's just make the correction and mark as a breaking change

Reply:
Sure, do I need to change this PR title's prefix to `fix!:`?

Reply:
I would @mdelapenya thoughts?

Reply:
Is it possible we deprecate the wrong field, adding the new field, to avoid another breaking change? In our future work for v1, we can remove all those deprecations

Reply:
Having two fields is going to be very fragile, anything could be using the wrong one resulting in broken behaviour what's very hard to spot. I would just take the hit now, which would ensure everyone just fixes their broken references.

Reply:
> to split this particular fix out from the test of the typo fixes

Makes sense, thanks for it. @NathanBaulch could you do as Steven suggested and remove the Endpoint breaking change to a separate PR? 🙏

Reply:
Done.